### PR TITLE
Specialcase restore

### PR DESCRIPTION
**Pull request by Forgind**
Fixes #

### CODE REVIEW

File: src/MSBuild/LiveLogger/LiveLogger.cs [hunk 1]
```diff
@@ -234,20 +263,23 @@ private void eventSource_TaskFinished(object sender, TaskFinishedEventArgs e)
...
-            int id = e.BuildEventContext!.ProjectInstanceId;
-            if (!projects.TryGetValue(id, out ProjectNode? node))
-            {
-                return;
+                if (e is TaskCommandLineEventArgs)
+                {
+                    return;
+                }
+                // Get project id
+                int id = e.BuildEventContext!.ProjectInstanceId;
```
Comment (JanKrivanek) [inline src/MSBuild/LiveLogger/LiveLogger.cs:273]: I know this is just copied, but I'd be more defensive here

```
int id = (e.?BuildEventContext ?? BuildEventContext.Invalid).ProjectInstanceId;
```

**Review by JanKrivanek — APPROVED** [review]
Overall looks good to go - I have just couple minor comments for considering.

A oneliner description of the PR intent would be nice as well ;-)

File: src/MSBuild/LiveLogger/LiveLogger.cs [hunk 2]
```diff
@@ -134,16 +136,19 @@ private void eventSource_ProjectStarted(object sender, ProjectStartedEventArgs e
...
-                // Add project
-                ProjectNode node = new(e)
-                {
-                    ShouldRerender = true,
-                };
-                projects[id] = node;
-                UpdateFooter();
+                ShouldRerender = true,
+            };
+            projects[id] = node;
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/LiveLogger.cs:144]: Why allocate new over mutating the existing one in the reentrance case?

File: src/MSBuild/LiveLogger/LiveLogger.cs [hunk 3]
```diff
@@ -134,16 +136,19 @@ private void eventSource_ProjectStarted(object sender, ProjectStartedEventArgs e
...
+            UpdateFooter();
+
+            if (e.TargetNames?.Contains("Restore") == true)
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/LiveLogger.cs:147]: I think it's worth it to explicitly specify `OrdinalIgnoreCase` here even if netfx makes us do it the ugly way

```suggestion
            if (e.TargetNames?.IndexOf("Restore", StringComparison.OrdinalIgnoreCase) > 0)
```

File: src/MSBuild/LiveLogger/LiveLogger.cs [hunk 4]
```diff
@@ -134,16 +136,19 @@ private void eventSource_ProjectStarted(object sender, ProjectStartedEventArgs e
...
+            {
+                TerminalBuffer.IsRestoring = true;
+                restoringProjects++;
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/LiveLogger.cs:150]: I don't understand this counter--I expected something more like "remember the ID of the first project that starts `Restore` and be done when that project instance completes". Does that not work in some circumstance?

File: src/MSBuild/LiveLogger/TerminalBuffer.cs [hunk 5]
```diff
@@ -116,7 +117,10 @@ public static void Render()
                 overallBuildState == OverallBuildState.Warning ? ANSIBuilder.Formatting.ForegroundColor.Yellow :
                 ANSIBuilder.Formatting.ForegroundColor.White;
 
-            string text = $"MSBuild - Build in progress - {FinishedProjects} finished projects";
+            string text = IsRestoring ?
+                $"MSBuild - Restore in progress - {FinishedProjects} finished projects" :
+                $"MSBuild - Build in progress - {FinishedProjects} finished projects";
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/TerminalBuffer.cs:122]: Future item: I'd like the entrypoint project listed here, so the text becomes something like `Restoring foo.csproj` or `Building bar.sln`.

**Review by rainersigwald — COMMENTED** [review]
I didn't finish the full review because I tried this out and it doesn't look like it's working:

image

Note that RestoreTask is running but it's not in "restore mode".

**Comment by rainersigwald** [comment]
@Forgind should we close this since #8619 already has a special case?

**Pull request closed without merging**